protocol/contracts/amo/TpfAmo.sol

```diff
@@ -0,0 +1,547 @@
+pragma solidity ^0.8.4;
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "./interfaces/AMO__IPoolHelper.sol";
+import "./interfaces/AMO__IAuraBooster.sol";
+import "./interfaces/AMO__ITempleERC20Token.sol";
+import "./helpers/AMOCommon.sol";
+import "./interfaces/AMO__IAuraStaking.sol";
+
```

Comment:
What is the migration path for this if the logic needs updating? Just a simple

1. Pause this contract (for good)
2. Rollout the new version

Or would it need any other more complex migration?

Reply:
1/ pause contract
2/ `withdrawAllAndUnwrap` directly from `amoStaking`
3/ recover tokens
4/ deploy new contract. set new amo contract as operator on `amoStaking`